plugins/ufm_log_analyzer_plugin/src/loganalyze/log_analyzers/link_flapping_analyzer.py

```diff
@@ -0,0 +1,124 @@
+#
+# Copyright © 2013-2024 NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# This software product is a proprietary product of Nvidia Corporation and its affiliates
+# (the "Company") and all right, title, and interest in and to the software
+# product, including all associated intellectual property rights, are and
+# shall remain exclusively with the Company.
+#
+# This software product is governed by the End User License Agreement
+# provided with the software product.
+#
+from datetime import datetime
+import gzip
+from itertools import islice
+import os
+import re
+import shutil
+from typing import List
+import pandas as pd
+from utils.netfix.link_flapping import get_link_flapping
+from loganalyze.log_analyzers.base_analyzer import BaseImageCreator
+import loganalyze.logger as log
+FILE_NAME_PATTERN=r"^secondary_(5m|1h|1d|1w)_(\d{14})\.gz$"
+TIME_PATTERN="%Y%m%d%H%M%S"
+
+class LinkFlappingAnalyzer(BaseImageCreator):
+    def __init__(self, telemetry_samples_csv:List[str], dest_image_path:str):
+        super().__init__(dest_image_path)
+        self._telemetry_samples_csv = telemetry_samples_csv
+        self._mapped_samples = {"5m":{}, "1h":{}, "1d":{}, "1w":{}}
+        self._map_files_by_sampling_rate_and_date()
+        self._funcs_for_analysis={self.plot_link_flapping_last_week}
+
+    def _map_files_by_sampling_rate_and_date(self):
+        for sample_file in self._telemetry_samples_csv:
+            try:
+                file_name = os.path.basename(sample_file)
+                match = re.match(FILE_NAME_PATTERN, file_name)
+                if match:
+                    interval_time = match.group(1)
```

Comment:
There is no need. In order to find a match, the whole regex needs to be aligned.
Since the regex does not have any optional part (with ?), it means that if there is a match, it has to have the 2 capture groups in it.